src/main/java/com/groupbyinc/api/Query.java

```diff
@@ -1009,6 +1011,46 @@ public Query setLanguage(String language) {
     return this;
   }
 
+  /**
+   * @return The pre-filter expression on the query.
+   */
+  public String getPreFilterExpression() {
+    return preFilterExpression;
+  }
+
+  /**
+   * <code>
+   * Sets the pre-filter expression on the query and restricts the results based on the logical conditions defined by
+   * the expression. Fields targeted by the pre-filter expression must be refinable in the upload configuration.
+   *
+   * For example, if your records have a refinable field called "brand" and you wish to pre-filter the search for the brands
+   * "Samsung" and "Sony" your pre-filter expression would look like:
+   *
+   * `brand IN ["Samsung", "Sony"]`
+   *
+   * If you wanted to enhance this pre-filter and your records have a refinable field called "price", with a nested field called
```

Comment:
@JamesPace-GBI , side question:What happens if somebody tries to do a pre-filter on an non-refinable field?

Reply:
I think they'll end up getting a 500 Internal Server Error, as it manifests as some variant of an Elasticsearch error. They should hopefully only get these errors when they are building the expression for the first time, unless they change the upload configuration of course.